cypress/e2e/downloads.cy.ts

```diff
@@ -1,354 +1,309 @@
+        setMedia(player);
+        cy.get('.playkit-pre-playback-play-button').should('exist').click({force: true});
+        cy.get('[data-testid="download-overlay-button"]').should('not.exist');
       });
     });
-    it('should be blocked for DRM', () => {
-      return loadPlayer()
-        .then(() => {
-          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-          // @ts-ignore
-          cy.stub(player, 'getVideoElement').returns({mediaKeys: {}});
-          setMedia();
-          return player?.ready();
-        })
-        .then(() => {
-          expect(requestCount).to.equal(0);
-        });
-    });
-    it.skip('should be blocked for image entry', () => {
-      return loadPlayer()
-        .then(() => {
-          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-          // @ts-ignore
-          cy.stub(player, 'isImage').returns(true);
-          setMedia();
-          return player?.ready();
-        })
-        .then(() => {
-          expect(requestCount).to.equal(0);
-        });
-    });
-  });
-
-  describe('download xhr call', () => {
-    let requestUrl = '';
-
-    beforeEach(() => {
-      cy.intercept('**/playManifest/**', req => {
-        ++requestCount;
-        requestUrl = req.url;
-        req.reply({
-          fixture: 'video.mp4'
-        });
-      }).as('playManifest');
-    });
-
-    afterEach(() => {
-      requestUrl = '';
-    });
-
-    describe('cdn url', () => {
-      it('should use cdn url from config if cdn url is set', async () => {
-        await loadPlayerAndSetMedia();
-        expect(requestUrl.indexOf(PLAYER_CONFIG.provider.env.cdnUrl)).to.not.equal(-1);
-      });
-      it('should not make download xhr call if cdn url is not set', async () => {
-        await loadPlayerAndSetMedia({provider: {env: {cdnUrl: ''}}});
-        expect(requestCount).to.equal(0);
+    it('should not exist for DRM media', () => {
+      mockKalturaBe();
+      loadPlayer().then(player => {
+        cy.stub(player, 'getVideoElement').returns({mediaKeys: {}});
+        setMedia(player);
+        cy.get('.playkit-pre-playback-play-button').should('exist').click({force: true});
+        cy.get('[data-testid="download-overlay-button"]').should('not.exist');
       });
     });
-    describe('partner id', () => {
-      it('should contain partner id from config if partner id is set', async () => {
-        await loadPlayerAndSetMedia();
-        expect(requestUrl.indexOf(`p/${PLAYER_CONFIG.provider.partnerId}`)).to.not.equal(-1);
-      });
-      it('should not make download call if partner id is not set', async () => {
-        await loadPlayerAndSetMedia({provider: {env: {partnerId: ''}}});
-        expect(requestCount).to.equal(0);
+    it('should be visible for non-DRM vod', () => {
+      mockKalturaBe();
+      loadPlayerAndSetMedia().then(() => {
+        cy.get('.playkit-pre-playback-play-button').should('exist').click({force: true});
```

Comment:
any specific reason to use the button click instead of .play() api ?

Reply:
yes. i had to get an element that can assure the video started playing. it's the only way it worked for me, from all options i tried, this was done after consulting with @semarche-kaltura .